File: CSharp/1071. Greatest Common Divisor of Strings.cs
public class Solution
{
    public string GcdOfStrings(string str1, string str2)
    {
        // check if str1 + str2 is equal to str2 + str1
        if (!(str1 + str2).Equals(str2 + str1)) return "";

        // find the greatest common divisor of the lengths of str1 and str2
        int result = resolver(str1.Length, str2.Length);

        // return the substring of str1 from index 0 to result
        return str1.Substring(0, result);
    }

    // find the greatest common divisor
    private int resolver(int a, int b)
    {
        // base case
        if (b == 0) return a;

        // recursive case
        return resolver(b, a % b);
    }

}

File: JavaScript/1071. Greatest Common Divisor of Strings.js
/**
 * @param {string} str1
 * @param {string} str2
 * @return {string}
 */
var gcdOfStrings = function (str1, str2) {
    if ((str1 + str2) != (str2 + str1)) return "";

    var result = resolver(str1.length, str2.length);
    return str1.substring(0, result);
};

function resolver(a, b) {
    if (b === 0) return a;

    return resolver(b, a % b);
}